File: src/Clients/accountant2/src/lib/models/entities/automaticTransaction.ts
import type { Syncable } from '$lib/models/syncable';

export class AutomaticTransaction implements Syncable {
	categoryName: string | undefined;
	synced = false;

	constructor(
		public id: number,
		public isDeposit: boolean,
		public categoryId: number | null,
		public amount: number,
		public currency: string,
		public description: string | null,
		public dayInMonth: number,
		public createdDate: Date | null,
		public modifiedDate: Date | null
	) {}
}

File: src/Domain/Accountant/AutomaticTransaction.cs
namespace Domain.Accountant;

public class AutomaticTransaction : Entity
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public bool IsDeposit { get; set; }
    public int? CategoryId { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; }
    public string Description { get; set; }
    public short DayInMonth { get; set; }
}
